src/aap_eda/tasks/orchestrator.py

```diff
@@ -333,7 +334,13 @@ def get_least_busy_queue_name() -> str:
             "No healthy queue found to dispatch the request",
         )
 
-    return queue_counter.most_common()[-1][0]
+    min_count = queue_counter.most_common()[-1][1]
+    least_common = [
+        queue for queue, count in queue_counter.items() if count == min_count
+    ]
+    if len(least_common) == 1:
+        return least_common[0]
+    return random.choice(least_common)
```

Comment:
Is checking the length really going to be significantly faster than just calling random.choice regardless?

Reply:
Sure, maybe the difference is minor but for sure it will be faster.
In OCP and single node scenarios there will be always one queue, so it is worth IMO.

Reply:
¯\_(ツ)_/¯

```
In [2]: l = ["value"]

In [3]: import random

In [4]: def no_len(l):
   ...:     return random.choice(l)
   ...: 

In [5]: def with_len(l):
   ...:     if len(l) == 1:
   ...:         return l[0]
   ...:     return random.choice(l)
   ...: 

In [6]: %timeit no_len(l)
230 ns ± 7.37 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [7]: %timeit with_len(l)
45.5 ns ± 1.45 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

Reply:
Ooooh, 6 ns!!!  Thank the deity all we ever do is billions of these specific operations. 😄